core/test/unit/api/v2/utils/validators/input/pages_spec.js

```diff
@@ -137,48 +137,52 @@ describe('Unit: v2/utils/validators/input/pages', function () {
         });
 
         describe('field formats', function () {
-            const fieldMap = {
-                title: [123, new Date(), _.repeat('a', 2001)],
-                slug: [123, new Date(), _.repeat('a', 192)],
-                mobiledoc: [123, new Date()],
-                feature_image: [123, new Date(), 'random words'],
-                featured: [123, new Date(), 'abc'],
-                status: [123, new Date(), 'abc'],
-                locale: [123, new Date(), _.repeat('a', 7)],
-                visibility: [123, new Date(), 'abc'],
-                meta_title: [123, new Date(), _.repeat('a', 301)],
-                meta_description: [123, new Date(), _.repeat('a', 501)]
-            };
-
-            Object.keys(fieldMap).forEach((key) => {
-                it(`should fail for bad ${key}`, function () {
-                    const badValues = fieldMap[key];
-
-                    const checks = badValues.map((value) => {
-                        const page = {};
+            let fieldMap,badValues,checks,page,frame, key;
+            before(function () {
+                fieldMap = {
+                    title: [123, new Date(), _.repeat('a', 2001)],
+                    slug: [123, new Date(), _.repeat('a', 192)],
+                    mobiledoc: [123, new Date()],
+                    feature_image: [123, new Date(), 'random words'],
+                    featured: [123, new Date(), 'abc'],
+                    status: [123, new Date(), 'abc'],
+                    locale: [123, new Date(), _.repeat('a', 7)],
+                    visibility: [123, new Date(), 'abc'],
+                    meta_title: [123, new Date(), _.repeat('a', 301)],
+                    meta_description: [123, new Date(), _.repeat('a', 501)]
+                };
+            }); 
+
+            beforeEach(function () {
+                Object.keys(fieldMap).forEach((key) => {
```

Comment:
Same here https://github.com/TryGhost/Ghost/pull/11154/files#r358046650. 
Thinking we might just create separate `it` statement per `key`. Could be something like:
```js
it(`should fail for bad slug`, function () {
    badValues = fieldMap.slug;
    checks = badValues.map((value) => {
        page = {};
        page[key] = value;

        frame = {
            options: {},
            data: {
                pages: [page]
            }
        };
        return validators.input.pages.add(apiConfig, frame)
            .then(Promise.reject)
            .catch((err) => {
                (err instanceof common.errors.ValidationError).should.be.true();
            });
    });
});
```

Or if you could think of a better approach? :)

Reply:
I'll stick to you're solution, thank you!